File: java Practice/QuickSort.java
public class QuickSort {
    public static void main(final String args[]) {

        final int arr[] = { 2, 22, 58, 88, 1, 5 };

        int[] output = new int[arr.length];

        final int left = 0, right = arr.length;

        output = QuickSort.Sort(arr, left, right);

        for (int i : output) {
            System.out.println(i);
        }

    }

    public static int pivot(final int arr[], int start, int end) {
        final int pivot = arr[start];
        int swapIndx = start;
        for (int i = start + 1; i < end; i++) {
            if (pivot > arr[i]) {

                swapIndx++;
                swap(arr, swapIndx, i);
            }
        }
        swap(arr, start, swapIndx);

        return swapIndx;

    }

    private static void swap(final int[] arr, final int i, final int j) {

        final int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;

    }

    public static int[] Sort(final int[] arr, int left, int right) {

        if (left < right) {

            final int pivotindex = pivot(arr, left, right);
            Sort(arr, left, pivotindex - 1);
            Sort(arr, pivotindex + 1, right);

        }

        return arr;
    }
}

File: js Practice/quickSort.js
pivot = (arr, start = 0, end = arr.length + 1) => {
  swap = (array, i, j) => {
    var temp = array[i];
    array[i] = arr[j];
    array[j] = temp;
  };
  let pivot = arr[start];
  let swapIdx = start;
  for (var i = start + 1; i < arr.length; i++) {
    if (pivot > arr[i]) {
      swapIdx++;
      swap(arr, swapIdx, i);
    }
  }
  swap(arr, start, swapIdx);

  return swapIdx;
};

quickSort = (arr, left = 0, right = arr.length - 1) => {
  if (left < right) {
    let pivotidx = pivot(arr, left, right);
    quickSort(arr, left, pivotidx - 1);
    quickSort(arr, pivotidx + 1, right);
  }
  return arr;
};
